### PR TITLE
[sound-manager][ACR-1147] Add guide for new device state

### PR DESCRIPTION
**Pull request by sangchul1011**
Signed-off-by: Sangchul Lee <sc11.lee@samsung.com>

### Change Description ###
Add new APIs to get device state. The running state of a device represents whether an audio stream is routed to the device or not. User can notice the state by using these new APIs.

### API Changes ###
ACR-1147

### CODE REVIEW

**Comment by journey2w** [comment]
@annie-samsung 
Please review english.

File: docs/application/native/guides/multimedia/sound-manager.md [hunk 1]
```diff
@@ -192,21 +198,59 @@ To query sound device information:
...
+              ret = sound_manager_get_device_type(device, &type);
+              if (type == SOUND_DEVICE_BLUETOOTH_MEDIA)
+                  /* Disconnected sound device type is Bluetooth Media, handle accordingly */
+              else
+                  /* Handle accordingly */
+          }
+      }
+      ```
+
+3. To receive a notification whenever the sound device running state changes:
```
Comment (Reni-90) [inline docs/application/native/guides/multimedia/sound-manager.md]: Suggestion:
Can we rephrase to:
To receive a notification whenever the sound device changes its running state:

Please verify whether the technical meaning is altered.

File: docs/application/native/guides/multimedia/sound-manager.md [hunk 2]
```diff
@@ -143,6 +143,7 @@ To query sound device information:
       - `sound_manager_get_device_io_direction()`: To get the device IO direction.
       - `sound_manager_get_device_id()`: To get the device ID.
       - `sound_manager_get_device_name()`: To get the device name.
+      - `sound_manager_is_device_running()`: To check if the device is running which means any stream is running on the device.
```
Comment (Reni-90) [inline docs/application/native/guides/multimedia/sound-manager.md]: Suggestion:
Can we rephrase to:
To verify whether the device is running. This means that a stream is running on the device.

Please verify whether the technical meaning is altered..

File: docs/application/native/guides/multimedia/sound-manager.md [hunk 3]
```diff
@@ -192,21 +198,59 @@ To query sound device information:
...
+
+   1. Register a callback using the `sound_manager_add_device_running_changed_cb()` function. Use the mask to filter the callback information.
+
+      ```
+      mask = SOUND_DEVICE_IO_DIRECTION_OUT_MASK | SOUND_DEVICE_IO_DIRECTION_BOTH_MASK;
+
+      ret = sound_manager_add_device_running_changed_cb(mask, _sound_device_running_changed_cb, NULL);
+      ```
+
+      > **Note**
```
Comment (Reni-90) [inline docs/application/native/guides/multimedia/sound-manager.md]: Please follow the Note tag style:

> **Note**
>  
> The state will be changed to **running** when the first stream goes to the device.

File: docs/application/native/guides/multimedia/sound-manager.md [hunk 4]
```diff
@@ -152,11 +153,16 @@ To query sound device information:
       while ((_ret = sound_manager_get_next_device(list, &device)) == SOUND_MANAGER_ERROR_NONE) {
           ret = sound_manager_get_device_type(device, &type);
 
-          if (type == SOUND_DEVICE_BLUETOOTH)
-              /* Sound device type is Bluetooth, handle accordingly */
+          if (type == SOUND_DEVICE_BLUETOOTH_MEDIA)
+              /* Sound device type is Bluetooth Media, handle accordingly */
```
Comment (Reni-90) [inline docs/application/native/guides/multimedia/sound-manager.md]: /* Sound device type is Bluetooth Media, handle accordingly */
What  we mean by "handle accordingly"? Please clarify.

File: docs/application/native/guides/multimedia/sound-manager.md (hunk 2 above)
Comment (sangchul1011) [reply docs/application/native/guides/multimedia/sound-manager.md]: It looks fine, will be updated soon.

File: docs/application/native/guides/multimedia/sound-manager.md (hunk 3 above)
Comment (sangchul1011) [reply docs/application/native/guides/multimedia/sound-manager.md]: Okay.

File: docs/application/native/guides/multimedia/sound-manager.md (hunk 4 above)
Comment (sangchul1011) [reply docs/application/native/guides/multimedia/sound-manager.md]: Honestly I don't know who put this phrase - handle accordingly - in all if-else statements.
Rather I would like to remove this pharse.

File: docs/application/native/guides/multimedia/sound-manager.md (hunk 1 above)
Comment (sangchul1011) [reply docs/application/native/guides/multimedia/sound-manager.md]: There are two type of device state.
 - Connection state : connected, disconnected.
 - Running state : running, not running(idle).

Both states are changed by user action(connect/disconnect a device and start/stop a stream respectively). Therefore the device state change is of passive, not of active.

From this point of view, is your suggestion still valid?
(if valid, do I need to change "whenever the sound device connection state changes" to "whenever the sound device changes its connection state"?)

File: docs/application/native/guides/multimedia/sound-manager.md (hunk 4 above)
Comment (Reni-90) [reply docs/application/native/guides/multimedia/sound-manager.md]: I agree with you.

File: docs/application/native/guides/multimedia/sound-manager.md (hunk 1 above)
Comment (sangchul1011) [reply docs/application/native/guides/multimedia/sound-manager.md]: It looks good to me, thanks.

File: docs/application/native/guides/multimedia/sound-manager.md (hunk 3 above)
Comment (Reni-90) [reply docs/application/native/guides/multimedia/sound-manager.md]: I am sorry, if I was not clear. The note tag style is:
> ** Note**
>
> Details to be mentioned.

Comment (sangchul1011) [reply docs/application/native/guides/multimedia/sound-manager.md]: Do you mean one more "empty line" between Note and the sentence?
or two spaces right after Note'xx'? (if the latter is the case, those are already there)

Comment (Reni-90) [reply docs/application/native/guides/multimedia/sound-manager.md]: Not only a space, a space with ">" tag.

> **Note**
>
> Details

**Comment by mijin85cho** [comment]
Hello, @Reni-90 
I can see that sangchul1011 updated this PR according to your suggestions.
Please check the updated version and approve it if you see no issues, so that I can merge this PR.

**Comment by Reni-90** [comment]
Hello @mijin85cho ,
Sorry for the delay. 
The changes are updated according to the suggestions. 
@sangchul1011,
Thank you so much.

**Comment by mijin80lee** [comment]
TD Sync Completed.(for Tizen_5.0_prepare)

**Pull request merged**